presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java

```diff
@@ -0,0 +1,537 @@
+
+        ensureValuesCapacity(positionCount, presentStream != null && nullsAllowed);
+
+        int streamPosition;
+
+        if (dataStream == null && lengthVector == null && isNullVector != null) {
+            streamPosition = readAllNulls(positions, positionCount);
+        }
+        else if (filter == null) {
+            streamPosition = readNoFilter(positions, positionCount);
+        }
+        else {
+            streamPosition = readWithFilter(positions, positionCount);
+        }
+
+        readOffset = offset + streamPosition;
+        return outputPositionCount;
+    }
+
+    private int readNoFilter(int[] positions, int positionCount)
+            throws IOException
+    {
+        int streamPosition = 0;
+        allNulls = false;
+
+        int offsetVectorIndex = 0;
+        offsetVector[offsetVectorIndex] = 0;
+        boolean isCharType = streamDescriptor.getOrcType().getOrcTypeKind() == OrcType.OrcTypeKind.CHAR;
+        int maxCodePointCount = getMaxCodePointCount(streamDescriptor.getOrcType());
+
+        for (int i = 0; i < positionCount; i++) {
+            int position = positions[i];
+            if (position > streamPosition) {
+                skipData(streamPosition, position - streamPosition);
+                streamPosition = position;
+            }
+
+            if (isNullVector != null && isNullVector[position]) {
+                offsetVector[offsetVectorIndex + 1] = offsetVector[offsetVectorIndex];
+                offsetVectorIndex++;
+                nulls[i] = true;
+            }
+            else {
+                int offset = offsetVector[offsetVectorIndex];
+                int length = lengthVector[lengthIndex];
+                int truncatedLength = 0;
+                if (length > 0) {
+                    dataStream.next(data, offset, offset + length);
+                    truncatedLength = computeTruncatedLength(slice, offset, length, maxCodePointCount, isCharType);
+                }
+                offsetVector[offsetVectorIndex + 1] = offset + truncatedLength;
+                offsetVectorIndex++;
+                lengthIndex++;
+                if (nullsAllowed && isNullVector != null) {
+                    nulls[i] = false;
+                }
+            }
+            streamPosition++;
+        }
+        outputPositionCount = positionCount;
+        outputPositions = positions;
+        return streamPosition;
+    }
+
+    private int readWithFilter(int[] positions, int positionCount)
+            throws IOException
+    {
+        int streamPosition = 0;
+        allNulls = false;
+        offsetVector = new int[positionCount + 1];
+
+        int offsetVectorIndex = 0;
+        offsetVector[offsetVectorIndex] = 0;
+        boolean isCharType = streamDescriptor.getOrcType().getOrcTypeKind() == OrcType.OrcTypeKind.CHAR;
+        int maxCodePointCount = getMaxCodePointCount(streamDescriptor.getOrcType());
+        int failedLengthCheckSize = 0;
```

Comment:
failedLengthCheckSize -> dataToSkip

Reply:
ok, in your previous comment you called it failedLengthCheckSize so changed it form dataToSkip to this. :)